filecoin-proofs/src/types/porep_config.rs

```diff
@@ -78,8 +78,8 @@ impl PoRepConfig {
         porep_id: [u8; 32],
         api_version: ApiVersion,
         api_features: Vec<ApiFeature>,
-    ) -> Self {
-        Self {
+    ) -> Result<Self> {
```

Comment:
Mentioning this just in case you prefer it; an early idea for `ApiFeature` was to have a way to check for conflicting features in `storage_proofs_core::api_version`. Based on your changes, I think that it would look something like...
```rust
impl ApiFeature {
    fn conflicting_features(&self) -> std::slice::Iter<Self> {
        match self {
            ApiFeature::SyntheticPoRep => [ApiFeature::NonInteractivePoRep].iter(),
            ApiFeature::NonInteractivePoRep => [ApiFeature::SyntheticPoRep].iter(),
            // _ => [].iter(),
        }
    }
}

pub fn check_conflicting_api_features(enabled_feats: &[ApiFeature]) -> Result<()> {
    let n = enabled_feats.len();
    if n < 2 {
        return Ok(());
    }
    for i in 0..n - 1 {
        let feat = &enabled_feats[i];
        let rem = &enabled_feats[i + 1..];
        for conflict in feat.conflicting_features() {
            if rem.contains(conflict) {
                return Err(anyhow!(
                    "The features `{feat}, {conflict}` cannot be enabled simultaneously"
                ));
            }
        }
    }
    Ok(())
}
```

```rust
impl PoRepConfig {
    pub fn new_groth16_with_features(
        sector_size: u64,
        porep_id: [u8; 32],
        api_version: ApiVersion,
        mut api_features: Vec<ApiFeature>,
    ) -> Result<Self> {
        api_features.dedup();
        storage_proofs_core::api_version::check_conflicting_api_features(&api_features)?;
        let partitions = if api_features.contains(&ApiFeature::SyntheticPoRep) {
            // Synth-PoRep partition count
        } else if api_features.contains(&ApiFeature::NonInteractivePoRep) {
            // NI-PoRep partition count
        } else {
            // Interactive-PoRep partition count
        };
        Ok(PoRepConfig {
            sector_size: SectorSize(sector_size),
            partitions,
            porep_id,
            api_version,
            api_features,
        })
    }
}
```

Reply:
I'm not saying that I prefer the above, but it does have the benefit of specifying conflicting features in the same file as `enum ApiFeature` (and where the features' `ApiVersion` ranges are specified).

Reply:
That sounds like a good idea, as it would "scale" better in case there are more features in the future. Let me try that out.

Reply:
Alright, I've pushed a new commit, which is a combination of what I had an your idea.